src/casp/services/api/services/cellarium_general_service.py

```diff
@@ -69,3 +70,26 @@ def get_model_for_user(self, model_name: str, user: models.User) -> models.CASMo
                 f"{model_name} model is not available. Please reach out to the Cellarium team for more information."
             )
         return model
+
+    def get_quota_for_user(self, user: models.User) -> schemas.UserQuota:
+        """
+        Get user quota information
+
+        :param user: User object to check quota for
+
+        :return: User quota information
+        """
+
+        days_to_add = datetime.datetime.now().weekday() - user.created_at.weekday()
+        if days_to_add <= 0:
+            days_to_add += 7
```

Comment:
I'm not really sure.  The logic right now for determining the start of the quota period would definitely break if we were to change that number.  Like, if we wanted to change the quota period to 2 weeks or 3 days or something, it would take more changes than just updating that number.

Reply:
That's a great point